branca/element.py

```diff
@@ -57,11 +57,13 @@ def __init__(self, template=None, template_name=None):
         self._env = ENV
         self._children = OrderedDict()
         self._parent = None
+        self.template = template
+        self.template_name = template_name
```

Comment:
I'm thinking whether it's better to make these protected attributes. To signal to users they are not meant for them. And to possibly avoid confusion between `_template` and `template` here.

How about  perhaps`self._template_str` and `self._template_name`?

Reply:
Done !